sympy/functions/elementary/trigonometric.py

```diff
@@ -2677,6 +2677,8 @@ def taylor_term(n, x, *previous_terms):
     def _eval_as_leading_term(self, x, logx=None, cdir=0):  # atan
         arg = self.args[0]
         x0 = arg.subs(x, 0).cancel()
+        if x0 is S.NaN:
```

Comment:
The other 2 times that PoleError is raised in this file there is a message in the error that tells what failed. Can a similar message be added in this case? What specifically just happened that caused the PoleError to raise?